jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarExtractor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.tar;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+
+/** Extracts a tarball. */
+public class TarExtractor {
+
+  /**
+   * Extracts a tarball to the specified destination.
+   *
+   * @param source the tarball to extract
+   * @param destination the output directory
+   * @throws IOException if extraction fails
+   */
+  public static void extract(Path source, Path destination) throws IOException {
+    try (InputStream in = Files.newInputStream(source)) {
+      try (TarArchiveInputStream tarArchiveInputStream = new TarArchiveInputStream(in)) {
```

Comment:
You can use a single try-resource.

```java
try (InputStream in = ...;
   TarArchiveInputStream ...) {
```

And in this case, I think using `BufferedInputStream` is meaningful. `BufferedInputStream` may loop to do multiple `read()` before returning, so it could reduce the number of `out.write()` if this happens.

Reply:
Should we use a `BufferedOutputStream` as well?

Reply:
I think it doesn't hurt, although we may be not so bad without it because we are not calling `write()` frequently with small bits.